go.mod

```diff
@@ -7,6 +7,7 @@ require (
 	github.com/google/go-cmp v0.7.0
 	github.com/google/uuid v1.6.0
 	github.com/stretchr/testify v1.11.1
+	go.uber.org/mock v0.6.0
```

Comment:
I assume this addition and the `go.sum` one were unintended? I don't see any usages

Reply:
Found the culprit, an outdated `segments_mock.go` 👍